File: leetcode-04/solutions/cpp/0769.cpp
class Solution {
 public:
  int maxChunksToSorted(vector<int>& arr) {
    int ans = 0;
    int max = INT_MIN;

    for (int i = 0; i < arr.size(); i++) {
      max = std::max(max, arr[i]);
      if (max == i) ans++;
    }

    return ans;
  }
};

File: leetcode-04/solutions/python3/0769.py
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        ans = 0
        max_ = float('-inf')

        for i, a in enumerate(arr):
            max_ = max(max_, a)
            if max_ == i:
                ans += 1

        return ans
